schema/parser.cpp

```diff
@@ -1396,6 +1396,7 @@ bool Parser::parse(ParserContext *context, QXmlInputSource *source)
 
 bool Parser::parseFile(ParserContext *context, QFile &file)
 {
+    context->setDocumentBaseUrlFromFileUrl(QUrl::fromLocalFile(file.fileName()));
```

Comment:
Should this maybe be done only if there's no base URL yet?
KDSoap supports setting it from the command line (via Settings::self()->wsdlBaseUrl()) and this would overwrite it, I think?

Reply:
If my search was correct in the KDSoap source there is not any call to the parseFile method. You are using the parseSchemaTag in the kdwsdl2cpp/wsdl//type.cpp:
https://github.com/KDAB/KDSoap/blob/master/kdwsdl2cpp/wsdl/type.cpp#L78

However I saw this and started thinking further:
https://github.com/cornelius/libkode/blob/master/schema/parser.cpp#L131

To remain on a safe side of the things would not be easier to set the setDocumentBaseUrlFromFileUrl in the kode/kxml_compiler code before calling the Parser::parseFile?

What do you think @winterheart